CrazyCanvas/Source/ECS/Systems/Player/WeaponSystemClient.cpp

```diff
@@ -81,34 +162,21 @@ void WeaponSystemClient::FixedTick(LambdaEngine::Timestamp deltaTime)
 	}
 }
 
-void WeaponSystemClient::Fire(EAmmoType ammoType, LambdaEngine::Entity weaponEntity)
+void WeaponSystemClient::Fire(LambdaEngine::Entity weaponEntity, WeaponComponent& weaponComponent, EAmmoType ammoType, const glm::vec3& position, const glm::vec3& velocity, uint32 playerTeam)
 {
 	using namespace LambdaEngine;
 
-	WeaponSystem::Fire(ammoType, weaponEntity);
-
-	ECSCore* pECS = ECSCore::GetInstance();
-	const WeaponComponent&		weaponComponent		= pECS->GetConstComponent<WeaponComponent>(weaponEntity);
-	const PositionComponent&	positionComponent	= pECS->GetConstComponent<PositionComponent>(weaponComponent.WeaponOwner);
-	const VelocityComponent&	velocityComponent	= pECS->GetConstComponent<VelocityComponent>(weaponComponent.WeaponOwner);
+	WeaponSystem::Fire(weaponEntity, weaponComponent, ammoType, position, velocity, playerTeam);
 
 	// Play gun fire and spawn particles
 	ISoundEffect3D* m_pSound = ResourceManager::GetSoundEffect3D(m_GunFireGUID);
```

Comment:
I know you didn't do this but this shouldn't be named m_pSound
```suggestion
	ISoundEffect3D* pSound = ResourceManager::GetSoundEffect3D(m_GunFireGUID);
```